main.tf

```diff
@@ -119,3 +124,133 @@ resource "aws_s3_bucket_policy" "this" {
   bucket = aws_s3_bucket.this.id
   policy = data.aws_iam_policy_document.this.json
 }
+
+########################################
+# Source replication configuration
+########################################
+resource "aws_s3_bucket_replication_configuration" "this" {
+  count = "${var.dest_region != "" ? 1 : 0}"
+
+  role   = aws_iam_role.replication[0].arn
+  bucket = aws_s3_bucket.this.id
+
+  rule {
+    id       = random_id.replication[0].b64_std
+    priority = 0
+
+    delete_marker_replication {
+      status = "Enabled"
+    }
+
+    filter {
+      prefix = ""
+    }
+
+    status = "Enabled"
+
+    destination {
+      bucket = aws_s3_bucket.destination[0].arn
+    }
+  }
+}
+
+########################################
+# Replicated bucket
+########################################
+resource "aws_s3_bucket" "destination" {
+  count    = "${var.dest_region != "" ? 1 : 0}"
+  provider = aws.destination
+
+  bucket = "${local.bucket_name}-replica"
```

Comment:
By default, this is `${data.aws_caller_identity.current.account_id}-${data.aws_region.region.name}-${var.name_suffix}-replica`..
which seems weird?

Reply:
Yeah, I went back and forth on this but I think this makes sense. I could update the region in the new bucket name but then it could potentially get confusing if there were a lot of replicated buckets. I figured keeping the name the same and adding -replica was the cleanest solution. I could argue for either way though.

Reply:
makes sense to me.. I don't feel strongly enough to say it should be changed and after reading your comment, I'm not sure which way I feel at all